Makefile

```diff
@@ -143,8 +143,17 @@ prow-ci-manifests-check: manifests
 		exit 1; \
 	fi
 
+.PHONY: vendor
+vendor:
+	for d in . dpu-api api tools ; do \
+		if [ "$$d" = . ] ; then \
+			(cd $$d && go mod vendor) || exit $$? ; \
+		fi ; \
+		(cd $$d && go mod tidy) || exit $$? ; \
+	done
+
 .PHONY: generate
-generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+generate: vendor controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
```

Comment:
This prevents separate commits with vendor

Reply:
how do you suggest to solve this?

The commit message explains the problem, and also has a reproducer that you can just run.

also, this commit is part of https://github.com/openshift/dpu-operator/pull/204 , which is why this PR is still a Draft.

Reply:
discussion moved to https://github.com/openshift/dpu-operator/pull/204#issuecomment-2491456950